src/binder/inc/bindertracing.h

```diff
@@ -49,6 +50,61 @@ namespace BinderTracing
         ReleaseHolder<PEAssembly> m_resultAssembly;
         bool m_cached;
     };
+
+
+    class ResolutionAttemptedOperation
+    {
+    private:
+        enum class Result : uint16_t
+        {
+            Attempt,
+            Success,
+            AssemblyNotFound,
+            MismatchedAssemblyName,
+            IncompatibleVersion,
+            Unknown,
+        };
+
+    public:
+        enum class Stage : uint16_t
+        {
+            NotYetStarted,
+            FindInLoadContext,
+            ALCLoad,
+            PlatformAssemblies,
+            DefaultALCFallback,
+            AppDomainAssemblyResolveEvent,
+        };
+
+        struct ResolutionAttempt
+        {
+            SString AssemblyName;
+            uint16_t Stage;
+            uint16_t Result;
+            SString ResultAssemblyName;
+            SString ResultAssemblyPath;
+            SString ErrorMessage;
+            SString AssemblyLoadContext;
+        };
+
+        ResolutionAttemptedOperation(BINDER_SPACE::AssemblyName *assemblyName);
+        ~ResolutionAttemptedOperation();
+
+        void SetResult(HRESULT hr, BINDER_SPACE::Assembly *assembly);
+
+        void Trace(Stage s, INT_PTR managedALC=0, AssemblySpec *assemblySpec=nullptr);
+
+    private:
+        BINDER_SPACE::AssemblyName *m_pAssemblyName;
+        Stage m_stage{Stage::NotYetStarted};
```

Comment:
Nit: can you move the value init to the ctor? I find it weird to have to look at two places (files) to determine the initial values of fields - especially since there is an explicit ctor.
Applies to all the other fields below as well.

Reply:
Done.